### ISSUE TITLE
Not authorized to commit changes from branch

### ISSUE THREAD

**Issue by lisamsmith100** [ga-wdi-boston/game-project#520]
My issue is that I am unable to push my committed changes to master branch.  I'd like to know and understand what commands I need to attach my feature branch to my repo and then I'd like to push so that I can try to deploy (despite nonworking site).

It appears my feature branch was not connected to my repo and I'm not sure why.  I have made quite a few changes on the feature branch, and have committed them.  

I discovered this change with the command: 'git push origin feature'.

When I review the repo, it appears only a branch exists for game logic.

Here are my bash commands/responses:

-------
$ git branch
* feature
~/wdi/projects/tic-tac-toe-client (feature)
$ git push origin feature
fatal: 'origin' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
-------

Thank you.

**Comment by awoodrum87** [comment]
did you initialize git in that repo?

**Comment by awoodrum87** [comment]
or did you fork it before cloning it?

**Comment by lisamsmith100** [comment]
I did.  I have a .git in there.

**Comment by lisamsmith100** [comment]
I forked, cloned and git inited.  I'm afraid to try to make the repair on my own for fear that I will lose my changes.  Hoping for some suggestions/insight.  Thanks for the chat, Angie.

**Comment by tvcrawley** [comment]
What do you get when you type `git remote -v`?

**Comment by lisamsmith100** [comment]
$ git remote -v
~/wdi/projects/tic-tac-toe-client (feature)
$

**Comment by lisamsmith100** [comment]
Updating status:  talked with Ben and we pushed changes to master.  Then attempted to deploy. 

Seeing this:

~/wdi/projects/tic-tac-toe-client (master)
$ git status
On branch master
Your branch is up-to-date with 'master/master'.
nothing to commit, working tree clean
~/wdi/projects/tic-tac-toe-client (master)
$ grunt deploy
grunt-cli: The grunt command line interface (v1.2.0)

Fatal error: Unable to find local grunt.

If you're seeing this message, grunt hasn't been installed locally to
your project. For more information about installing and configuring grunt,
please see the Getting Started guide:

http://gruntjs.com/getting-started
~/wdi/projects/tic-tac-toe-client (master)
$ git branch
  feature
* master
~/wdi/projects/tic-tac-toe-client (master)
$

**Comment by lisamsmith100** [comment]
I am noticing that my changes are not displayed in the master but are displayed in the feature branch.  I'm contemplating doing the following, knowing this is not normal way of doing things:

1. copying and pasting my code from feature into master.
2. attempting to redeploy and hope for success.
2. creating a new branch by cloning from master, never to touch the feature branch again.
3. making future changes in branches OTHER THAN FEATURE.

Is this OK for troubleshooting today?

**Comment by BenGitsCode** [comment]
I'll come take a look at this with you.

**Comment by BenGitsCode** [comment]
...after lunch 😅  Please find or ping me when you return.

**Comment by lisamsmith100** [comment]
Thank you to both Bens for their time and care with the help.

I'm still not entirely sure what happened  to create the issue, but the situation was one where the feature branch on which I was working was not a child branch of master and therefore could not push to master or deploy master.

At first, we created a remote connection to master, but the changes were not pushed.
When that did not work, we tried something different and were successful.
We used the following commands (pasted from terminal):

541  git remote add master git@github.com:lisamsmith100/tic-tac-toe-client.git
  542  git status
  543  git branch
  544  git push origin master
  545  git remote
  546  git branch
  547  git push master
  548  git push --set-upsteam master feature
  549  git status
  550  git push --set-upstream master feature
  551  git status
  552  git branch
  553  git branch
  554  git add assets/scripts/game/logic.js
  555  git add assets/styles/tacoImage.png
  556  git add assets/styles/towTruck.jpg
  557  git status
  558  git commit
  559  git status
  560  git push origin feature
  561  git push master
  562  git pull master
  563  git status
  564  git checkout --assets/scripts/gameEvents/game.js
  565  git status
  566  git checkout -- assets/scripts/gameEvents/game.js
  567  git status
  568  git pull master
  569  git status
  570  git branch
  571  grunt deploy
  572  git status
  573  grunt deploy
  574  git branch
575  git branch
  576  git checkout feature
  577  git branch
  578  git status
  579  git add assets/scripts/game/events.js
  580  git status
  581  git commit
  582  git status
  583  git branch
  584  git remote -v
  585  git push origin feature
  586  git remote rm master
  587  git remote -v
  588  git remote add origin git@github.com:lisamsmith100/tic-tac-toe-client.git
  589  git fetch --all
  590  git branch
  591  git branch --all
  592  git branch -D gameLogic
  593  git push origin :gameLogic
  594  git status
  595  git checkout master
  596  git status
  597  atom .
  598  git diff origin/feature --name-status
  599  git push orgin master
  600  git push oirgin master
  601  git push origin master
  602  git checkout feature
  603  git push origin feature
  604  git reflog
  605  git branch -m
  606  git branch -b feature-new
  607  git branch
  608  git status
  609  git checkout -b newfeature
  610  git checkout master
  611  git branch -m master feature
  612  git branch -D feature
  613  git checkout -b newmaster
  614  git branch -D master
  615  git status
  616  git branch
  617  git checkout newfeature
 618  git branch -m master
  619  git status
  620  git branch
  621  git log
  622  git branch -D newmaster
  623  git status
  624  git branch
  625  git push origin master
  626  git log
  627  git log --oneline
  628  git push origin master --force-with-lease
  629  history
~/wdi/projects/tic-tac-toe-client (master)
$    
$ git checkout -b newfeature
Switched to a new branch 'newfeature'
~/wdi/projects/tic-tac-toe-client (newfeature)
$ git status
On branch newfeature
nothing to commit, working tree clean
~/wdi/projects/tic-tac-toe-client (newfeature)
$ 

I'm on a new branch and am glad all of my commits were saved and we have a clean branch.  

Thank you again for your help!
